janus_core/calculations/md.py

```diff
@@ -499,6 +509,80 @@ def __init__(
 
         self._parse_correlations()
 
+    def _set_time_step(self):
+        """Set time in fs and current dynamics step to info."""
+        time = (self.offset * self.timestep + self.dyn.get_time()) / units.fs
+        step = self.offset + self.dyn.nsteps
+        self.dyn.atoms.info["time_fs"] = time
+        self.dyn.atoms.info["step"] = step
+
+    def _prepare_restart(self) -> None:
+        """Prepare restart files, structure and offset."""
+        # Check offset can be read from steps
+        try:
+            with open(self.stats_file, encoding="utf8") as stats_file:
+                last_line = stats_file.readlines()[-1].split()
+            try:
+                self.offset = int(last_line[0])
+            except (IndexError, ValueError) as e:
+                raise ValueError("Unable to read restart file") from e
+        except FileNotFoundError as e:
+            raise FileNotFoundError("Unable to read restart file") from e
```

Comment:
Do we need this nested? The outer one would probably handle all cases, particularly since the error messages are the same.

Reply:
I'd probably prefer to distinguish between the two messages better, rather than compress it and potentially leave it more ambiguous what the cause of the error was.

Reply:
The context will come from `e` just as it does here.

Reply:
Yes, but a better message like "Unable to infer step from last line of restart file" makes the problem more immediately clear than a generic message with the IndexError/ValueError context and needing to follow the trace to understand what caused them

Reply:
E.g. at the moment, the output is essentially:

```
│    496 │   │                                                                                     │
│    497 │   │   self.offset = 0                                                                   │
│    498 │   │   if self.restart:                                                                  │
│ ❱  499 │   │   │   self._prepare_restart()                                                       │
│    500 │   │   self.restart_files = []                                                           │
│    501 │   │   self.created_final_file = False                                                   │
│    502                 

│    525 │   │   │   try:                                                                          │
│    526 │   │   │   │   self.offset = int(last_line[0])                                           │
│    527 │   │   │   except (IndexError, ValueError) as e:                                         │
│ ❱  528 │   │   │   │   raise ValueError("Unable to read restart file") from e                    │
│    529 │   │   except FileNotFoundError as e:                                                    │
│    530 │   │   │   raise FileNotFoundError("Unable to read restart file") from e                 │
│    531      

ValueError: Unable to read restart file
```

If we get rid of the differentiation, I get something more like:

```
│    496 │   │                                                                                                                                                           │
│    497 │   │   self.offset = 0                                                                                                                                         │
│    498 │   │   if self.restart:                                                                                                                                        │
│ ❱  499 │   │   │   self._prepare_restart()                                                                                                                             │
│    500 │   │   self.restart_files = []                                                                                                                                 │
│    501 │   │   self.created_final_file = False                                                                                                                         │
│    502         

│    526 │   │   │   self.offset = int(last_line[0])                                                                                                                     │
│    527 │   │                                                                                                                                                           │
│    528 │   │   except (FileNotFoundError, IndexError, ValueError) as e:                                                                                                │
│ ❱  529 │   │   │   raise FileNotFoundError("Unable to read restart file") from e                                                                                       │
│    530 │   │                                                                                                                                                           │
│    531 │   │   if self.restart_auto:                                                                                                                                   │
│    532 │   │   │   # Attempt to infer restart file

FileNotFoundError: Unable to read restart file
```

So actually neither is particularly enlightening at the moment, but at least the current version is a bit more specific

Reply:
I suppose, to me that doesn't make that much of a difference, but I can see where you're coming from. Leave as-is.